### PR TITLE
Build: Enable debugging for java-ci

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Too many times, I am facing one flaky error (also other people are facing it too). Unable to know the reason. Hence, enabling the `--stacktrace`, `--info` for future debugging.  

```
* What went wrong:
Execution failed for task ':iceberg-spark:iceberg-spark-3.3_2.12:checkstyleMain'.
> A failure occurred while executing org.gradle.api.plugins.quality.internal.CheckstyleAction
   > An unexpected error occurred configuring and executing Checkstyle.
      > java.lang.Error: Error was thrown while processing /home/runner/work/iceberg/iceberg/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java
```

Failure details:
https://github.com/apache/iceberg/actions/workflows/java-ci.yml

https://github.com/apache/iceberg/actions/runs/4309428443
https://github.com/apache/iceberg/actions/runs/4310045826
https://github.com/apache/iceberg/actions/runs/4306392493/
and many more

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @Fokko, @nastra, @jackye1995

**Review by jackye1995 — APPROVED** [review]

File: .github/workflows/java-ci.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ jobs:
       with:
         distribution: zulu
         java-version: 8
-    - run: ./gradlew -DallVersions build -x test -x javadoc -x integrationTest
+    - run: ./gradlew -DallVersions build -x test -x javadoc -x integrationTest --scan
```
Comment (XN137) [inline .github/workflows/java-ci.yml:90]: doesnt this require to agree to gradle's terms of service?
see https://docs.gradle.com/enterprise/gradle-plugin/#connecting_to_scans_gradle_com

imo if the sporadic errors always happen in checkstyle part, we may want to check if we can upgrade the respective plugin to fix the issue

Comment (jackye1995) [reply .github/workflows/java-ci.yml:90]: > doesnt this require to agree to gradle's terms of service
good point, in that case we probably cannot enable it

Comment (ajantha-bhat) [reply .github/workflows/java-ci.yml:90]: I think other projects like Nessie Agree to these terms of service. I have skimmed it and didn't find anything harmful.

Do you guys think we should not enable `--scan` and just enable `--stacktrace` `--debug `?

Comment (ajantha-bhat) [reply .github/workflows/java-ci.yml:90]: >  imo if the sporadic errors always happen in checkstyle part, we may want to check if we can upgrade the respective plugin to fix the issue

we are using the latest version I assume as I didn't find any version pinning. 
https://github.com/apache/iceberg/blob/master/baseline.gradle#L36

Comment (XN137) [reply .github/workflows/java-ci.yml:90]: we are only using latest version of baseline which supported java 8:
https://github.com/apache/iceberg/blob/35692a360ffd2d1b9107ec6018f9fd97fda04671/build.gradle#L30-L34

in there we can find the checkstyle version:
https://github.com/palantir/gradle-baseline/blob/4.42.0/versions.props#L17

but i think 9.3 would be the last version with java8 support:
https://checkstyle.org/releasenotes.html#Release_9.3

Comment (jackye1995) [reply .github/workflows/java-ci.yml:90]: I think it's more of a licensing issue, not sure if there is any other project under Apache Foundation using this feature, we can ping the infra team about this.

Comment (ajantha-bhat) [reply .github/workflows/java-ci.yml:90]: True but the `gradle-baseline-4.42.0` may not accept new patches :( 

Also what is your opinion on 
> Do you guys think we should not enable --scan and just enable --stacktrace --debug ?

Comment (ajantha-bhat) [reply .github/workflows/java-ci.yml:90]: we need to have the failure reason before raising the patch to the gradle-baseline. So, I believe we need to get stack trace/debug first.

Comment (XN137) [reply .github/workflows/java-ci.yml:90]: > before raising the patch to the gradle-baseline

i was trying to say there would probably be ways to bump the checkstyle version without needing to get a new gradle-baseline version

that being said, i agree we should try to get stacktrace first to see what exactly goes wrong occasionally

**Comment by nastra** [comment]
Rather than merging this PR with debugging enabled, I'd rather vote for rerunning CI on this PR until we run into the issue, as otherwise this will be spamming logs. That being said, I'm leaning towards -1 on merging this.

**Comment by ajantha-bhat** [comment]
changed to `--info` now. I will rerun the PR until I face the problem

**Comment by ajantha-bhat** [comment]
no luck in reproducing yet!

**Comment by ajantha-bhat** [comment]
Discovered a new flaky test 🤦 
https://github.com/apache/iceberg/issues/6988

**Comment by ajantha-bhat** [comment]
I rebased and pushed to this branch when the PR was closed. So, I can't reopen the PR. will open a new PR again to debug this.

**Pull request closed without merging**